Lamina/src/GL/texture.cpp

```diff
@@ -0,0 +1,67 @@
+#include "../Lamina/include/GL/texture.hpp"
+
+lm::Texture2D::Texture2D(std::string path)
+{
+	LoadFromFile(&path[0]);
+}
+
+int lm::Texture2D::LoadFromFile(const char* path)
+{
+	textureData = stbi_load(path, &width, &height, &channels, 0);
+	glActiveTexture(GL_TEXTURE0);
+	glGenTextures(1, &texture);
+	glBindTexture(GL_TEXTURE_2D, texture);
+	if (textureData)
+	{
+		glTexParameteri(GL_TEXTURE_2D, GL_TEXTURE_WRAP_S, GL_REPEAT);
+		glTexParameteri(GL_TEXTURE_2D, GL_TEXTURE_WRAP_T, GL_REPEAT);
+		glTexParameteri(GL_TEXTURE_2D, GL_TEXTURE_MIN_FILTER, GL_LINEAR);
+		glTexParameteri(GL_TEXTURE_2D, GL_TEXTURE_MAG_FILTER, GL_LINEAR);
+
+		if (channels == STBI_grey) glTexImage2D(GL_TEXTURE_2D, 0, GL_RED, width, height, 0, GL_RED, GL_UNSIGNED_BYTE, textureData);
+		else if (channels == STBI_grey_alpha) glTexImage2D(GL_TEXTURE_2D, 0, GL_RG, width, height, 0, GL_RG, GL_UNSIGNED_BYTE, textureData);
+		else if(channels == STBI_rgb) glTexImage2D(GL_TEXTURE_2D, 0, GL_RGB, width, height, 0, GL_RGB, GL_UNSIGNED_BYTE, textureData);
+		else if (channels == STBI_rgb_alpha) glTexImage2D(GL_TEXTURE_2D, 0, GL_RGBA, width, height, 0, GL_RGBA, GL_UNSIGNED_BYTE, textureData);
+		else return 0;
```

Comment:
This will lead to a memory leak since `textureData` clearly exists but the function returns before it is freed, at which point the pointer is lost and it's impossible to free the memory afterwards.